src/CSI.IBTA.ProcessingService/Services/ScopedProcessingService.cs

```diff
@@ -0,0 +1,95 @@
+        IBenefitsUnitOfWork _benefitsUnitOfWork) : IScopedProcessingService
+    {
+        public async Task DoWorkAsync(CancellationToken stoppingToken)
+        {
+            while (!stoppingToken.IsCancellationRequested)
+            {
+                if (_logger.IsEnabled(LogLevel.Information))
+                {
+                    _logger.LogInformation("Worker running at: {time}", DateTimeOffset.Now);
+                }
+
+                var currentDate = DateTime.UtcNow.Date;
+
+                var enrollments = await _benefitsUnitOfWork.Enrollments.GetSet()
+                    .Include(e => e.Plan.Package)
+                    .Where(e => e.Plan.Package.Initialized != null)
+                    .Where(e => e.Plan.Package.PlanStart <= currentDate && e.Plan.Package.PlanEnd >= currentDate)
+                    .ToListAsync();
+
+                foreach (Enrollment enrollment in enrollments)
+                {
+                    var transactionCount = await _benefitsUnitOfWork.Transactions.GetSet()
+                        .Where(t => t.EnrollmentId == enrollment.Id)
+                        .Where(t => t.Type == TransactionType.Income)
+                        .CountAsync(cancellationToken: stoppingToken);
+
+                    Plan plan = enrollment.Plan;
+                    Package package = plan.Package;
+                    DateTime startDate = package.PlanStart;
+                    DateTime endDate = package.PlanEnd;
+                    PayrollFrequency payrollFrequency = package.PayrollFrequency;
+
+                    int payrollFrequencyDays = payrollFrequency == PayrollFrequency.Weekly ? 7 : 30;
+
+                    int numberOfPays = CalculateNumberOfPayments(payrollFrequencyDays, startDate, endDate, DateTime.UtcNow);
+                    int transactionCountDifference = numberOfPays - transactionCount;
+
+                    decimal totalInsuranceAmount = plan.Contribution + enrollment.Election;
+                    int totalNumberOfPays = CalculateNumberOfPayments(payrollFrequencyDays, startDate, endDate, endDate);
+                    decimal transactionAmount = totalInsuranceAmount / totalNumberOfPays;
+
+                    // Handles situations if system was down and missed payments
+                    for (int i = 0; i < transactionCountDifference; i++)
+                    {
+                        var transaction = new Transaction
+                        {
+                            Amount = transactionAmount,
+                            DateTime = DateTime.UtcNow,
+                            Enrollment = enrollment,
+                            Type = TransactionType.Income
+                        };
+
+                        await _benefitsUnitOfWork.Transactions.Add(transaction);
+                    }
+                }
+
+                await _benefitsUnitOfWork.CompleteAsync();
+
+                await Task.Delay(WorkerConfigConstants.CHECK_INTERVAL_MS, stoppingToken);
```

Comment:
Usually timers are used for the repeatable actions. As you already emulated it with Task.Delay, not actions required. 
https://learn.microsoft.com/en-us/dotnet/api/system.timers.timer?view=net-8.0 
https://learn.microsoft.com/en-us/dotnet/api/system.threading.timer?view=net-8.0&redirectedfrom=MSDN